liferay-echo/liferay-portal
# LPS-163083 Proof of concept - DM browser tag

# Motivation

This implements a tag that will allow other applications (must be portlets) to embed a simplified view of documents and media with limited functionality.

This is still a proof of concept, so error handling leaves much to be desired and there are little things here and there that are missing/wrong:

- Success/error messages are not translated.
- Only card view is supported.
- No empty state in the search container.

Currently, the tag supports four basic operations:

1. Navigate through the folder structure.
2. Create folders.
3. Upload a single file.
4. Delete folders and files.

# Steps to Verify

As this is a POC, there's no need to review the code, as it will change quite a bit in future iterations. Your main job is to test that the tag doesn't make your app explode when used.

To use your tag in one of your portlets, you need to do the following:
```
<liferay-document-library:repository-browser repositoryId="<%= ... %>" />
```
where `repositoryId` is the repository you want to display. This attribute is optional; if you don't provide a value, the default repository will be displayed.

For your convenience, I've already included an usage of this tag in the Blogs Images tab. There, you should see the contents of Documents and Media.

![image](https://user-images.githubusercontent.com/729897/194537593-0084aedd-6a6f-4ac0-95d0-adc4f687bb03.png)

Thanks!
